client/src/App.js

```diff
@@ -38,29 +38,45 @@ function App() {
     return library.find((book) => book.id === focusedBook.id);
   }
 
-  function renderBookDetails(book) {
-    return (
-      <BookDetails
-        book={book}
-        onRemoveDetailView={() => setView('')}
-        onAddRating={addRating}
-      />
-    );
-  }
-
   function renderBookDetailsHelper(book) {
     setDetailedBook(book);
     setView('details');
   }
 
   function updateBook(property, value, bookToUpdate) {
-    const upDatedBooks = library.map((book) => {
+    const updatedBooks = library.map((book) => {
       if (book.id === bookToUpdate.id) {
         book[property] = value;
       }
       return book;
     });
-    setLibrary(upDatedBooks);
+    setLibrary(updatedBooks);
+  }
+
+  function updateBooksInCompartment(property, selection, book) {
+    const updatedShelves = shelves.map((shelf) => {
+      if (shelf.id === selection.bookshelf) {
```

Comment:
To make it clear that it's both ids that you are comparing, you could rename `selection.bookshelf` to `selection.bookshelfId`